### PR TITLE
Migrate memory_heap_treemap.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart [hunk 1]
```diff
@@ -26,37 +24,33 @@ class MemoryHeapTreemap extends StatefulWidget {
...
-  InstructionsSize sizes;
-
-  Map<String, Function> callbacks = {};
+  MemoryHeapTreemapState(this._controller);
 
-  MemoryController controller;
+  InstructionsSize? _sizes;
 
-  Widget snapshotDisplay;
+  MemoryController? _controller;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart:31]: late and non nullable

File: packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart [hunk 2]
```diff
@@ -26,37 +24,33 @@ class MemoryHeapTreemap extends StatefulWidget {
...
-  TreemapNode root;
+  TreemapNode? root;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
 
     // TODO(terry): Unable to short-circuit need to investigate why?
-    controller = Provider.of<MemoryController>(context);
+    _controller = Provider.of<MemoryController>(context);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart:40]: use the `initialized` pattern

File: packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart [hunk 3]
```diff
@@ -244,31 +235,30 @@ class InstructionsSize {
...
-  final String className;
+  final String? libraryUri;
+  final String? className;
 
   List<String> get parts {
     return <String>[
-      if (libraryUri != null) ...libraryUri.split('/') else '@stubs',
-      if (className != null && className.isNotEmpty) className,
+      if (libraryUri != null) ...libraryUri!.split('/') else '@stubs',
+      if (className?.isNotEmpty == true) className!,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart:261]: nit: `className?.isNotEmpty ?? false`

File: packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart:31]: done

File: packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart:40]: done

File: packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart:261]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**